src/main/java/pages/BasePage.java

```diff
@@ -0,0 +1,38 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+
+public class BasePage {
+    public WebDriver driver;
+    private By singInLink = By.className("authorization-link");
+    private By welcomeSpan = By.className("logged-in");
+    private By menOption = By.className("level0 nav-3 category-item level-top parent ui-menu-item");
+    private By topsOption = By.id("ui-id-17");
+    private By jacketOption = By.className("level0 nav-3 category-item level-top parent ui-menu-item");
+
+    public BasePage (WebDriver driver) {this.driver = driver;}
+
+    public void clickSingin(){
+        driver.findElement(singInLink).click();
+    }
+    public void hoverMenOption(){
+        Actions action = new Actions(driver);
```

Comment:
For the handling of the "Actions" type elements, I suggest handling it the same as the "Webdriver" type element, so we avoid having to instantiate the same element several times:

```
public Actions action;

public BasePage (WebDriver driver, Actions action){
this.driver = driver;
this.action = action;
}
```
and then use it in the methods